src/Components/PaymentMethod/PaymentMethodContainer.js

```diff
@@ -316,6 +316,54 @@ const PaymentMethodContainerWithoutStripe = ({
     debounce(onApplyCouponCode, 1000)
   ).current;
 
+  const removeAppliedCoupon = (state) => {
+    state.couponCode = "";
+
+    dispatch({ type: SET_COUPON_ERROR, payload: "" });
+
+    dispatch({
+      type: SHOW_COUPON_FIELD,
+      payload: false
+    });
+
+    dispatch({
+      type: SET_COUPON,
+      payload: null
+    });
+
+    dispatch({
+      type: SET_PERCENT_OFF,
+      payload: ""
+    });
+
+    dispatch({
+      type: SET_UPDATED_PRICE,
+      payload: null
+    });
+
+    dispatch({
```

Comment:
nitpick: instead of dispatching all these actions, I think we should've added a new action type that handles all these changes